src/tests/mod.rs

```diff
@@ -47,28 +50,46 @@ fn initialize() {
 /// * dest: Path to destination
 ///
 /// This method causes side effects - writes new files to a directory
-fn copy_folder(src: &Path, dst: &Path) {
-  for entry in fs::read_dir(src).unwrap() {
+fn copy_folder_to_temp_dir(src: &PathBuf) -> TempDir {
+  // Copy the test scenario to temporary directory
+  let temp_dir = TempDir::new_in(".", "tmp_test").unwrap();
+  let temp_dir_path = &temp_dir.path();
+  for entry in fs::read_dir(src.as_path()).unwrap() {
     let entry = entry.unwrap();
     let path = entry.path();
     if path.is_file() {
       _ = fs::copy(
         path.to_str().unwrap(),
-        dst.join(path.file_name().unwrap()).to_str().unwrap(),
+        temp_dir_path
+          .join(path.file_name().unwrap())
+          .to_str()
+          .unwrap(),
       )
       .unwrap();
     }
   }
+  temp_dir
 }
 
 /// Checks if the file updates returned by piranha are as expected.
-fn check_result(path_to_codebase: &Path, path_to_expected: &Path) {
+fn execute_piranha_and_check_result(
+  piranha_arguments: &PiranhaArguments, path_to_expected: &Path, files_changed: usize,
+  ignore_whitespace: bool,
+) {
+  let path_to_codebase = Path::new(piranha_arguments.path_to_codebase());
+
+  let output_summaries = execute_piranha(piranha_arguments);
+
+  assert!(output_summaries.iter().any(|x| !x.rewrites().is_empty()));
```

Comment:
What is this line checking for? Maybe worth a comment about what this check means?

There was a comment before: `// Checks if there are any rewrites performed for the file`, though I'd use the opportunity to explain it more clearly. Is the check here to make sure that at least one file was changed? (i.e. that the test as a whole actually resulted in a rewrite) Why do we need this test in addition to the next one, then? (Not saying we don't, just not clear to me from reading the code)

Reply:
I had this check before we had updated all of this code. I guess now it seems a bit redundant.
It also restricts us to test scenarios where we always expect a rewrite. Maybe I want to add a test case where I dont expect a rewrite :) 
I am going to just remove this.